### PR TITLE
[ASTS] Targeted Sweeper Part 4: Separating out the single batch sweeper

### PR DESCRIPTION
**Pull request by mdaudali**
## General
**Before this PR**:
See the previous 3 parts. This time splitting out the sweeper, to make testing a little bit easier for the old design.

**After this PR**:
Factors out the single batch sweeper.

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**:
N/A
**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No
**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No
**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes
**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No
**Does this PR need a schema migration?**
No
## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
No change
**What was existing testing like? What have you done to improve it?**:
Using existing test suite at this time
**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A
**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A
## Execution
No change

If a problem, rollback
## Scale

No change

## Development Process
**Where should we start reviewing?**:
SingleBatchSweeper
**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:
N/A
**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SingleBatchSweeper.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+public interface SingleBatchSweeper {
+    long sweepNextBatch(ShardAndStrategy shardStrategy, long sweepTs);
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SingleBatchSweeper.java:20]: this is just temporary right? / the foreground task needs SweepableBucket.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/DefaultSingleBatchSweeper.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        this.cleaner = cleaner;
+    }
+
+    /**
+     * Sweep the next batch for the shard and strategy specified by shardStrategy, with the sweep timestamp sweepTs.
+     * After successful deletes, the persisted information about the writes is removed, and progress is updated
+     * accordingly.
+     *
+     * @param shardStrategy shard and strategy to use
+     * @param sweepTs       sweep timestamp, the upper limit to the start timestamp of writes to sweep
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/DefaultSingleBatchSweeper.java:57]: not actionable: I didn't think we normally aligned this, but certainly no objections if we wish to do so

**Review by jeremyk-91 — COMMENTED** [review]
just one question to check my understanding of the broader strategy

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SingleBatchSweeper.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SingleBatchSweeper.java:20]: checked offline: yeah, this is not intended to also be used for the foreground task in any way

**Review by jeremyk-91 — APPROVED** [review]
👍

**Comment by mdaudali** [comment]
👍

**Pull request merged**